c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -0,0 +1,434 @@
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+int FSteam::ValidateInputAndReturnValidatedInput(float Input, std::string Message, int LowerBound, int UpperBound)
+{
+	while (!std::cin.good() || !IsInteger(Input) || !IsInRange(Input, LowerBound, UpperBound))
+	{
+		ResetConsoleInputScreen();
+
+		std::cout << Message;
+		std::cin >> Input;
+	}
+	return (int)Input;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+std::string FSteam::ValidateInputAndReturnValidatedInput(std::string StringInput, std::string Message)
+{
+	while (!std::cin.good() || StringInput == "")
+	{
+		ResetConsoleInputScreen();
+
+		std::cout << Message;
+		std::cin >> StringInput;
+	}
+	return StringInput;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+int FSteam::GetValidDay(float TemporalDay, int Month, int Year)
+{
+	switch (Month)
+	{
+		// Case for february
+	case 2:
+		if (Year % 4 == 0 && Year != 1900)
+		{
+			return ValidateInputAndReturnValidatedInput(TemporalDay, "Please insert the actual day the game was published: ", 1, 29);
+		}
+		else
+		{
+			return ValidateInputAndReturnValidatedInput(TemporalDay, "Please insert the actual day the game was published: ", 1, 28);
+		}
+
+		// Cases of months that have 31 days
+	case 1:
+	case 3:
+	case 5:
+	case 7:
+	case 8:
+	case 10:
+	case 12:
+		return ValidateInputAndReturnValidatedInput(TemporalDay, "Please insert the actual day the game was published: ", 1, 31);
+
+		// Cases of months that have 30 days
+	case 4:
+	case 6:
+	case 9:
+	case 11:
+		return ValidateInputAndReturnValidatedInput(TemporalDay, "Please insert the actual day the game was published: ", 1, 30);
+
+	default:
+		"AddGame Date Error, please contact support";
+		return 1;
+	}
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::ResetConsoleInputScreen()
+{
+	std::cin.clear();
+	std::cin.ignore(INT_MAX, '\n');
+
+	system("cls");
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+FSteam::FSteam() : ActiveCommand(EMenuCommand::MainMenu), NumberOfGames(0)
+{
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::RunSteam()
+{
+	while (true)
```

Comment:
This is perfectly fine, but better reserved for cases where it's hard to specify the condition. In this case it's easy (when active command isn't exit) so it's better to be explicit about it.